### PR TITLE
Feature/1683 define eventlist format

### PR DESCRIPTION
**Pull request by randy-seng**
OP#1683

### CODE REVIEW

File: OTAnalytics/domain/event.py [hunk 1]
```diff
@@ -1,2 +1,156 @@
...
+
+    Raises:
+        ValueError: vehicle_id < 1
+        ValueError: frame_number < 1
+
+    Args:
+        road_user_id (int): the road user id involved with this event. It must be
+        greater equal one.
+        road_user_type (RoadUserType): the road user type involved with this event.
+        hostname (str): the OTCamera hostname that the video is associated with.
```
Comment (randy-seng) [inline OTAnalytics/domain/event.py:88]: How do we treat videos not recorded by an OTCamera?

File: OTAnalytics/domain/event.py [hunk 2]
```diff
@@ -1,2 +1,177 @@
...
+        raise InproperFormattedFilename(f"Could not parse {file_path.name}.")
+
+
+class SectionEventBuilder(EventBuilder):
+    def __init__(self) -> None:
+        self.section_id: Optional[str] = None
+        self.direction_vector: Optional[DirectionVector2D] = None
+        self.event_type: Optional[EventType] = None
+
+    def reset(self) -> None:
```
Comment (briemla) [inline OTAnalytics/domain/event.py:136]: Why do we need a reset method?
Can we not just create a new Builder?

File: OTAnalytics/domain/geometry.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    def _validate(self) -> None:
+        if len(self.coordinates) < 3:
+            raise ValueError(
+                (
+                    "Number of coordinates to define a valid polygon must be "
+                    f"greater equal three, but is {len(self.coordinates)}"
+                )
+            )
+
+        if self.coordinates[0] != self.coordinates[-1]:
```
Comment (briemla) [inline OTAnalytics/domain/geometry.py:40]: This means, we need at least 3 coordinates. If exaclty 3 are given, coordinate 1 and must be same. That would be a line. So the number of coordinates must either be greater or equal 4 or coordinates[0] will automatically added at the end to close the polygon.

Comment (randy-seng) [reply OTAnalytics/domain/geometry.py:40]: That is correct! Thank you! I'll fix that~

File: OTAnalytics/domain/event.py (hunk 2 above)
Comment (randy-seng) [reply OTAnalytics/domain/event.py:138]: I did it just in case some one tries to reuse the builder. But of course we can just use a new one

**Review by briemla — APPROVED** [review]

**Pull request merged**